source/auth/auth.rst

```diff
@@ -1159,6 +1159,199 @@ If AWS authentication fails for any reason, the cache MUST be cleared.
+    description: "Server's reply to clientStep1"
+    strict: false
+    fields:
+      authorizeEndpoint:
+        description: >-
+          URL where the IDP may be contacted for end user
+          authentication and authorization code generation.
+        type: string
+        optional: true # Req if deviceAuthorizeEndpoint not present
+      tokenEndpoint:
+        description: >-
+          URL where the IDP may be contacted for authorization
+          code <=> ID/access token exchange.
+        type: string
+        optional: true # Req if deviceAuthorizeEndpoint not present
+      deviceAuthorizeEndpoint:
+        description: >-
+          URL where the IDP may be contacted for device
+          authentication and authorization code generation.
+        type: string
+        optional: true # Req if authorizeEndpoint not present
+      clientId:
+        description: "Unique client ID for this OIDC client"
+        type: string
+      clientSecret:
+        description: "Secret used when communicating with IDP"
+        type: string
+        optional: true
+      requestScopes:
+        description: "Additional scopes to request from IDP"
+        type: array<string>
+        optional: true
+
+Server will use principalName (n) if provided in clientStep1 to select an appropriate IDP.  This IDP's configuration will be returned in serverStep1 to instruct the client on how to acquire an Access Token.  This Access Token will be used in clientStep2 to complete authentication.  The principalName identified within the Access Token MUST match any provided during clientStep1, or the authentication operation will fail.
+
+.. code:: idl
+
+  OIDCMechanismClientStep2:
+      description: "Client's request with signed token"
+      strict: false
+      fields:
+          jwt:
+              description: "Compact serialized JWT with signature"
+              cpp_name: JWT
+              type: string
+
+The driver MAY, if it has previously acquired a valid Access Token, or has cached the reply from serverStep1, begin their authentication by sending clientStep2 with their initial saslStart request and authenticate in a single round trip.  Similarly, as an authenticated client's token's expirationTime approaches, it should proactively acquire a new Access Token, and it MAY do so with cached IDP details, skipping directly to clientStep2.
+
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+    MUST NOT be specified.
+
+source
+    MUST be "$external". Defaults to ``$external``.
+
+password
+    MUST NOT be specified.
+
+mechanism
+    MUST be "MONGODB-OIDC"
+
+mechanism_properties
+    PRINCIPAL_NAME
+        Drivers MUST allow the user to specify a principal name, which
+        is required by the server when more than one Identity Provider
+        is configured.
+    DEVICE_NAME
+        Drivers MUST allow the user to specify a name for the device
+        workflow that is one of "aws", "azure", or "gcp".
```

Comment:
should it be case sensitive?

Reply:
I used lowercase to be consistent with how it is specified in kms